HIP/hip-844.md

```diff
@@ -0,0 +1,88 @@
+---
+hip: 844
+title: Handling and externalisation improvements for account nonce updates
+author: Stoyan Panayotov <stoyan.panayotov@limechain.tech>
+working-group: Nana Essilfie-Conduah <nana@swirldslabs.com>, Danno Ferrin <danno.ferrin@swirldslabs.com>, Steven Sheehy <steven.sheehy@swirldslabs.com>
+type: Standards Track
+needs-council-approval: Yes
+category: Service
+status: Review
+created: 2023-12-12
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/845
+---
+
+## Abstract
+
+Introduce rules for when to update an Ethereum transaction signer's nonce and a new field for externalising the nonce update in the record stream. This will help resolve issues where the Consensus nodes and the Mirror nodes are out of sync in regards to what the current value of an account's nonce is.
+
+## Motivation
+
+Unclear rules for when the nonce of an EthereumTransaction signer account should be updated result in discrepancies between the Consensus nodes' and Mirror nodes' states. This leads to users experiencing issues when reading the account nonce from the Mirror node and trying to use it in a transaction sent to the Consensus node.
```

Comment:
Q: Should we consider listing the error case from the HAPI side and relay that are symptomatic of the problem.
In this way some users might link the issue back to the resolution from this HIP.

Reply:
Added the error in the Backward Compatibility section.